### PR TITLE
Frontend checklist implementation

### PR DESCRIPTION
**Pull request by manuelmeister**
- Admin ui
- ContentNode Checklist

See example: https://bacluc-1.ecamp3.ch/
See feature toggle: https://pr5460.ecamp3.ch/camps/3c79b99ab424/GRGR/program/activities/12f34c89ce11

Frontend part of #4936

### CODE REVIEW

**Comment by manuelmeister** [comment]
## Core Meeting Decision

- Add feature toggle for admin ui, contentnode, navigation to disable this feature on prod

**Comment by BacLuc** [comment]
works like a charm, thank you

image

File: .helm/ecamp3/values.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ helpLink: # 'https://ecamp3.ch/faq'
 # enable/disable feature across the complete deployment
 featureToggle:
   developer: false # enables various tools/features foreseen for development deployments (language switcher, form controls view, performance measurement view, etc.)
+  checklist: false # enables various tools/features foreseen for development deployments (language switcher, form controls view, performance measurement view, etc.)
```
Comment (BacLuc) [inline .helm/ecamp3/values.yaml]:
```suggestion
  checklist: false
```
the comment is wrong.
thankfully, you did not attach the feature to the developer flag

File: .helm/ecamp3/values.yaml [hunk 2]
```diff
@@ -251,7 +252,7 @@ apiCache:
       requests:
         cpu: 10m
         memory: 20Mi
-  
+
```
Comment (BacLuc) [inline .helm/ecamp3/values.yaml:255]: unrelated

File: frontend/src/components/activity/ButtonNestedContentNodeAdd.vue [hunk 3]
```diff
@@ -117,6 +121,9 @@ export default {
         contentType.name !== 'ResponsiveLayout' || this.parentContentNode.parent === null
       )
     },
+    showChecklistNode(contentType) {
+      return contentType.name === 'Checklist' ? this.featureChecklistEnabled : true
```
Comment (BacLuc) [inline frontend/src/components/activity/ButtonNestedContentNodeAdd.vue]: i would have expected
```suggestion
      return this.featureChecklistEnabled && contentType.name === 'Checklist'
```

File: frontend/src/components/activity/content/Checklist.vue [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        this.contentNode
+          .checklistItems()
+          .items.some(({ _meta }) => _meta.self === item._meta.self)
+      )
+    },
+    serverSelection() {
+      return this.selectionContentNode.map((item) => item.id)
+    },
+    allChecklists() {
+      return this.camp.checklists().items.map((checklist) => ({
```
Comment (BacLuc) [inline frontend/src/components/activity/content/Checklist.vue]: should there be some null checks / coalesce operators here?
I fear some NPE's here

File: frontend/src/components/activity/content/Checklist.vue [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+  },
+  beforeDestroy() {
+    this.checkedItems = null
+    this.uncheckedItems = null
+  },
+  methods: {
+    // get all parents of a given item
+    getParents(item) {
+      const parents = []
+      let parent = item?.parent?.()
```
Comment (BacLuc) [inline frontend/src/components/activity/content/Checklist.vue:223]: what is happending here?

File: frontend/src/components/activity/content/Checklist.vue [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+      while (parent) {
```
Comment (BacLuc) [inline frontend/src/components/activity/content/Checklist.vue:224]: maybe you should copy the parent array, that you don't mutate shared state with unshift.

**Review by BacLuc — COMMENTED** [review]
some minor adjustments

File: frontend/src/components/activity/ButtonNestedContentNodeAdd.vue (hunk 3 above)
Comment (manuelmeister) [reply frontend/src/components/activity/ButtonNestedContentNodeAdd.vue]:
```suggestion
      return contentType.name !== 'Checklist' || this.featureChecklistEnabled
```

File: .helm/ecamp3/values.yaml (hunk 2 above)
Comment (manuelmeister) [reply .helm/ecamp3/values.yaml:255]: Sorry, can't force commit via IDE

File: frontend/src/components/activity/content/Checklist.vue [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+      checkedItems: [],
+      uncheckedItems: [],
+      errorMessages: null,
+      debouncedSave: () => null,
+      itemsLoaded: false,
+    }
+  },
+  computed: {
+    campChecklistItems() {
+      return this.api.get().checklistItems().items
```
Comment (BacLuc) [inline frontend/src/components/activity/content/Checklist.vue]: müsste man hier nicht noch bisschen filtern

File: frontend/src/components/activity/content/Checklist.vue (hunk 4 above)
Comment (BacLuc) [reply frontend/src/components/activity/content/Checklist.vue]: could be ok, if .item is always available

File: frontend/src/components/activity/content/Checklist.vue (hunk 5 above)
Comment (manuelmeister) [reply frontend/src/components/activity/content/Checklist.vue:223]: I need all parents to be able to display the path

File: frontend/src/components/activity/content/Checklist.vue (hunk 6 above)
Comment (manuelmeister) [reply frontend/src/components/activity/content/Checklist.vue:224]: This is not shared state

**Review by BacLuc — APPROVED** [review]

**Review by simfeld — APPROVED** [review]
- feature toggle works ✅
- creating and editing checklists ✅
- using checklists in schedule entries ✅
- drag & drop causes backend errors ❌ 

```
An exception occurred while executing a query: SQLSTATE[23505]:
Unique violation: 7
ERROR: duplicate key value violates unique constraint "checklistitem_checklistid_parentid_position_unique" DETAIL: Key (checklistid, parentid, "position")=(c5163ea9d9a6, 9d13b06a9080, 2) already exists. 
```

**Comment by simfeld** [comment]
Now also displays checklists if more than 1 of them exist :+1: LGTM
